h2/stream.py

```diff
@@ -1023,14 +1030,24 @@ def stream_reset(self, frame):
     def _build_headers_frames(self,
                               headers,
                               encoder,
-                              first_frame):
+                              first_frame,
+                              events):
```

Comment:
Rather than pass the events in here, can we hoist the building of the validation flags up a level? Maybe we need to write a helper function to do that, but that shouldn't be hard, and it avoids this method doing too much.